src/apple/target.rs

```diff
@@ -94,11 +95,21 @@ impl Reportable for BuildError {
 }
 
 #[derive(Debug)]
-pub struct ArchiveError(bossy::Error);
+pub enum ArchiveError {
+    SetVersionFailed(WithWorkingDirError<bossy::Error>),
+    ArchiveFailed(bossy::Error),
+}
 
 impl Reportable for ArchiveError {
     fn report(&self) -> Report {
-        Report::error("Failed to archive via `xcodebuild`", &self.0)
+        match self {
+            Self::SetVersionFailed(cause) => {
```

Comment:
The style I've stuck to for these sorts of 1-tuple error variants is `err` instead of `cause`